e2etest/pkg/frr/config/config.go

```diff
@@ -66,26 +76,32 @@ type NeighborConfig struct {
 
 // Set the IP of each node in the cluster in the BGP router configuration.
 // Each node will peer with the BGP router.
-func BGPPeersForAllNodes(cs clientset.Interface, nc NeighborConfig, rc RouterConfig) (string, error) {
+func BGPPeersForAllNodes(cs clientset.Interface, nc NeighborConfig, rc RouterConfig, ipFamily string) (string, error) {
 	router := rc
 
+	router.V4Neighbors = make([]*NeighborConfig, 0)
+	router.V6Neighbors = make([]*NeighborConfig, 0)
 	router.Neighbors = make([]*NeighborConfig, 0)
 
 	nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
 	if err != nil {
 		return "", errors.Wrapf(err, "Failed to get cluster nodes")
 	}
 
-	// TODO: The IP address handling will need updates to add support for dual stack
-	for _, node := range nodes.Items {
-		for i := range node.Status.Addresses {
-			if node.Status.Addresses[i].Type == "InternalIP" {
-				neighbor := nc
-				neighbor.Addr = node.Status.Addresses[i].Address
-				router.Neighbors = append(router.Neighbors, &neighbor)
-			}
+	ips := k8s.NodeIPsForFamily(nodes.Items, ipFamily)
+	for _, ip := range ips {
+		neighbor := nc
+		neighbor.Addr = ip
+		nc.IPFamily = k8s.IPFamilyForAddress(ip)
+
+		if nc.IPFamily == "ipv4" {
+			router.V4Neighbors = append(router.V4Neighbors, &neighbor)
+		} else {
+			router.V6Neighbors = append(router.V6Neighbors, &neighbor)
 		}
+		router.Neighbors = append(router.Neighbors, &neighbor)
```

Comment:
Hmm do we need to remove this line and put an error if its not v4 nor v6 ?

Reply:
See the comment above about the need for Neighbors